messaging-client-databus/src/main/java/com/inmobi/messaging/consumer/databus/DatabusConsumer.java

```diff
@@ -255,4 +292,60 @@ protected void createCheckpoint() {
       currentCheckpoint = new CheckpointList(partitionMinList);
     }
   }
+
+  @Override
+  public synchronized void close() {
+    newReaders.clear();
+    if (null != partitionDiscovererAndReaderCreator) {
+      partitionDiscovererAndReaderCreator.cancel();
+    }
+    super.close();
+  }
+
+  protected void startNewReaders() throws IOException{
+    for (PartitionId id : newReaders.keySet()) {
+      PartitionReader reader = newReaders.get(id);
+      reader.start(getReaderNameSuffix());
+      readers.put(id, reader);
+      LOG.info("started new reader " + getReaderNameSuffix() +
+          " " + reader.toString());
```

Comment:
yes. but different object hash codes can be used in debugging as well. not changing this. I actually used the hash code in debugging